src/streamlink/plugins/ceskatelevize.py

```diff
@@ -5,131 +5,168 @@
 $region Czechia
 """
 
-import logging
+import json
 import re
 
 from streamlink.plugin import Plugin, PluginError, pluginmatcher
 from streamlink.plugin.api import validate
 from streamlink.stream.dash import DASHStream
 
-log = logging.getLogger(__name__)
-
 
 @pluginmatcher(re.compile(
-    r"https://(?:www\.)?ceskatelevize\.cz/zive/\w+"
+    r"https?://(?:(?P<subdomain>\w+)\.)?ceskatelevize\.cz/(?:#live|zive/\w+)"
 ))
 class Ceskatelevize(Plugin):
-    _re_playlist_info = re.compile(r"{\"type\":\"([a-z]+)\",\"id\":\"([0-9]+)\"")
+    schema_playlist = {
+        "playlist": [{
+            "streamUrls": {
+                "main": validate.url(),
+            },
+        }],
+    }
 
-    def _get_streams(self):
-        self.session.http.headers.update({"Referer": self.url})
+    def get_stream_url(self, id):
```

Comment:
`id` shadows the built-in function. I usually avoid this, as it's showing up as a warning in pycharm via its default linting rules. Same below. Optional though, and just wanted to mention it.